test/integration/consul-container/libs/node/consul-container.go

```diff
@@ -0,0 +1,129 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/hashicorp/consul/api"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+)
+
+const bootLogLine = "Consul agent running"
+const disableRYUKEnv = "TESTCONTAINERS_RYUK_DISABLED"
+
+// consulContainerNode implement a Node
+// it instantiate Consul as a container
+type consulContainerNode struct {
+	ctx       context.Context
+	client    *api.Client
+	container testcontainers.Container
+	ip        string
+	port      int
+}
+
+// NewConsulContainer create a Node implemented as a consulContainerNode
+func NewConsulContainer(ctx context.Context, config Config) (Node, error) {
+
+	name := utils.RandName("consul-")
+	tmpDir, err := ioutils.TempDir("", name)
+	if err != nil {
+		return nil, err
+	}
+	err = os.Chmod(tmpDir, 0777)
+	if err != nil {
+		return nil, err
+	}
+	err = os.Mkdir(tmpDir+"/config", 0777)
+	if err != nil {
+		return nil, err
+	}
+	configFile := tmpDir + "/config/config.hcl"
+	err = os.WriteFile(configFile, []byte(config.HCL), 0644)
+	if err != nil {
+		return nil, err
+	}
+	skipReaper := isRYUKDisabled()
+	req := testcontainers.ContainerRequest{
+		Image:        "consul:" + config.Version,
+		ExposedPorts: []string{"8500/tcp"},
+		WaitingFor:   wait.ForLog(bootLogLine).WithStartupTimeout(10 * time.Second),
+		AutoRemove:   false,
+		Name:         name,
+		BindMounts:   map[string]string{"/consul/config/config.hcl": configFile},
```

Comment:
I think the original comment got outdated due to other changes. I think you are right though. It does look like the field got replaced with a generic `Mounts` field of type [`ContainerMounts`](https://pkg.go.dev/github.com/testcontainers/testcontainers-go#ContainerMounts).

Still, with new code I think it might be worth pulling in the latest version of the lib and making the updates so we are not immediately using deprecated/removed functionality.